ConfigSpace/forbidden.py

```diff
@@ -72,12 +74,13 @@ def is_forbidden(self, instantiated_hyperparameters):
 
 
 class AbstractForbiddenClause(AbstractForbiddenComponent):
-    def get_descendant_literal_clauses(self):
+    def get_descendant_literal_clauses(self) -> List['AbstractForbiddenClause']:
```

Comment:
Wouldn't `AbstractForbiddenComponent` be more appropriate her? Also, why is this a string? Moreover, the definition further below suggests that this function return `List[Hyperparameter].`